File: Java/src/com/sonicscholar/simpleBoardGame/mock/MockPoorlyImplementedAIPlayer.java
package com.sonicscholar.simpleBoardGame.mock;

import com.sonicscholar.simpleBoardGame.Board;
import com.sonicscholar.simpleBoardGame.GameAction;

/**
 * This class simulates a poorly implemented AI
 * player that does not return valid game actions.
 * It's used for testing that proper exceptions are
 * thrown so the programmer is made aware.
 */
public class MockPoorlyImplementedAIPlayer extends MockPlayer {
    public MockPoorlyImplementedAIPlayer(String name, char marker, Board board) {
        super(name, marker, board);
    }

    @Override
    public boolean isComputer() {
        return true;
    }

    @Override
    public GameAction requestAction() {
        return null;
    }
}

File: C_Sharp/SonicScholar.SimpleBoardGame.Framework/SonicScholar.SimpleBoardGame.Framework.Test/Mock/MockPoorlyImplementedAIPlayer.cs
namespace SonicScholar.SimpleBoardGame.Framework.Test.Mock
{
    /// <summary>
    /// This class simulates a poorly implemented AI
    /// player that does not return valid game actions.
    /// It's used for testing that proper exceptions are
    /// thrown so the programmer is made aware.
    /// </summary>
    class MockPoorlyImplementedAiPlayer : MockPlayer
    {
        public MockPoorlyImplementedAiPlayer(string name, char marker, IBoard board)
            : base(name, marker, board)
        {
        }

        public override bool IsComputer => true;

        /// <summary>
        /// Requests an action from a poorly implemented AI player
        /// </summary>
        /// <returns>null</returns>
        public override IGameAction RequestAction()
        {
            return null;
        }
    }
}
